### PR TITLE
Feature/score debugger Build Debugger

### PR DESCRIPTION
**Pull request by m4rtint**
Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs

Is really the only new thing

### CODE REVIEW

**Comment by mikeslgoh** [comment]
[(2) Create a score debugger for insta win](https://trello.com/c/hwiXOsZe/605-2-create-a-score-debugger-for-insta-win)

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
+﻿using _PartyParkade.Script.Data.Tools;
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:1]: Missing Header

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        DrawButton();
+    }
+    
+    #region IncrementPlayerScore
+    private void DrawTable()
+    {
+        GUILayout.BeginVertical();
+        GUILayout.BeginHorizontal();
+        // Header
+        var labelStyle = m_skin.GetStyle(m_label);
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:39]: replace vat with GUIStyle

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            HandleButton(player);
+            GUILayout.EndHorizontal();
+        }
+        
+        GUILayout.EndVertical();
+    }
+
+    private void HandleText(Player player)
+    {
+        var score = m_scoreDebugger.GetPlayerScore(player);
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:59]: replace var

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        GUILayout.BeginHorizontal();
+        GUILayout.Label(Constants.m_SliderLabel, m_skin.GetStyle(m_titleSkin));
+        HandleSlider();
+        GUILayout.TextField($"{m_scoreDebugger.GetWinningScore()}", m_skin.GetStyle(m_titleSkin));
+        GUILayout.EndHorizontal();
+    }
+
+    private void HandleSlider()
+    {
+        var currentWinningScore = m_scoreDebugger.GetWinningScore();
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:86]: var

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        currentWinningScore = (int) GUILayout.HorizontalSlider(currentWinningScore, 0, 20);
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:87]: maybe save 20 as a field, for easier changing later

File: Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs [hunk 6]
```diff
@@ -1,104 +1,117 @@
...
+        {
+            public static string m_PlayerColumnHeader = "Player";
+            public static string m_ScoreColumnHeader = "Score";
+            public static string m_IncrementColumnHeader = "Increment";
+            public static string m_SliderLabel = "Winning Score:";
+            public static string m_PlayerScore = "Score : {0}";
+            public static string m_InstantWinButtonText = "Insta Win";
+            public static string m_IncrementButtonText = "Increment Score";
+            public static float m_ExtraAreaTopMargin = 100;
+            public static int m_MaxScore = 20;
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Debug/Debug View/ScoreDebuggerView.cs:27]: if these are public should we remove the m_

**Comment by Gursaanj** [comment]
So the Score Debugger works when I use the Debug Window, but the Editor Window doesnt seem to be working 

image

Error Log
image

**Comment by m4rtint** [comment]
> So the Score Debugger works when I use the Debug Window, but the Editor Window doesnt seem to be working
> 
> image
> 
> Error Log
> image

I didn't write it for the Editor Window if that's what you're asking.

**Pull request merged**